protocol/communities_messenger_signers_test.go

```diff
@@ -128,102 +154,93 @@ func (s *MessengerCommunitiesSignersSuite) TestControlNodeUpdate() {
-		s.alice,
-		func(r *MessengerResponse) bool {
-			return len(r.Communities()) > 0 && r.Communities()[0].Description().Identity.Description == anotherJohnDescr
-		},
-		"no communities",
-	)
-	s.Require().ErrorContains(err, "no communities")
+	err = s.alice.AddCommunityToken(community.IDString(), int(chainID), tokenAddress)
+	s.Require().NoError(err)
 
-	// make john aware of control node change
-	_, err = s.john.communitiesManager.UpdateControlNode(community.ID(), bobPubKey)
+	// make alice the control node
+	transaction := transactions.SendTxArgs{}
+	_, err = s.alice.SetCommunitySignerPubKey(context.Background(), community.ID(), chainID, communityAddress, transaction, "password", common.PubkeyToHex(&s.alice.identity.PublicKey))
 	s.Require().NoError(err)
 
-	// john can't publish community update anymore
-	_, err = s.john.EditCommunity(&requests.EditCommunity{
-		CommunityID: community.ID(),
-		CreateCommunity: requests.CreateCommunity{
-			Name:        community.Name(),
-			Description: anotherJohnDescr,
-			Color:       community.Color(),
-			Membership:  protobuf.CommunityPermissions_NO_MEMBERSHIP,
-		},
-	})
-	s.Require().Error(err, communities.ErrNotEnoughPermissions)
+	var published bool
 
-	// FIXME: make community clock timestamp hinted lamport clock
-	_, err = s.bob.communitiesManager.IncreaseClock(community.ID())
-	s.Require().NoError(err)
+	// We check the updated description is published as a signal
+	signal.SetMobileSignalHandler(func(data []byte) {
+		response := &MessengerResponse{}
+		err := json.Unmarshal(data, response)
+		if err != nil {
+			s.john.logger.Error("failed to unmarshal json", zap.Error(err))
+			return
+		}
 
-	// bob as control node publishes community update
-	bobDescr := "bob's description"
-	response, err = s.bob.EditCommunity(&requests.EditCommunity{
-		CommunityID: community.ID(),
-		CreateCommunity: requests.CreateCommunity{
-			Name:        community.Name(),
-			Description: bobDescr,
-			Color:       community.Color(),
-			Membership:  protobuf.CommunityPermissions_NO_MEMBERSHIP,
-		},
+		published = strings.Contains(string(data), community.IDString())
 	})
-	s.Require().NoError(err)
-	s.Require().Equal(bobDescr, response.Communities()[0].Description().Identity.Description)
 
 	// john accepts community update
-	_, err = WaitOnMessengerResponse(
+	response, err = WaitOnMessengerResponse(
 		s.john,
 		func(r *MessengerResponse) bool {
-			return len(r.Communities()) > 0 && r.Communities()[0].Description().Identity.Description == bobDescr
+			return published
 		},
 		"no communities",
 	)
 	s.Require().NoError(err)
 
-	// alice accepts community update
+	// We check the control node is correctly set on john
+	johnCommunity, err := s.john.communitiesManager.GetByIDString(community.IDString())
+	s.Require().NoError(err)
+	s.Require().NotNil(johnCommunity)
+	s.Require().True(common.IsPubKeyEqual(johnCommunity.ControlNode(), &s.alice.identity.PublicKey))
+	s.Require().False(johnCommunity.IsControlNode())
```

Comment:
need to check, that Alice became an Owner, as the control node always has a role Owner
and john should loose his role and get it based on the community permission

Reply:
that is a bit less clear, we discussed with @osmaczko , but it's alice responsibility to then update the description and remove John from the member list as having token ownership (as I understand, there's some cases where we might want to have a separate owner from control node), this PR is not currently addressing that though (maybe in a separate?)

Reply:
Agree, not in this. I have PR with kicking all members